phpBB/phpbb/install/module_base.php

```diff
@@ -110,10 +110,38 @@ public function run()
 
 		if ($task_index < $iterator->count())
 		{
-			$iterator->seek($task_index);
+		    // Receiving task index
```

Comment:
I need to look at this when i have some more time, but it seems like something that would increase the execution time possibly quite a bit. 

As far as i understand you try to go back 1 stage with the obtain data stages, which would be fine but i think the ticket refers to restarts after a timeout.